stdlib/pathlib/_abc.pyi

```diff
@@ -0,0 +1,128 @@
+    @overload
+    def split(self, path: PathLike[AnyStr]) -> tuple[AnyStr, AnyStr]: ...
+    @overload
+    def split(self, path: AnyOrLiteralStr) -> tuple[AnyOrLiteralStr, AnyOrLiteralStr]: ...
+    @overload
+    def splitdrive(self, path: PathLike[AnyStr]) -> tuple[AnyStr, AnyStr]: ...
+    @overload
+    def splitdrive(self, path: AnyOrLiteralStr) -> tuple[AnyOrLiteralStr, AnyOrLiteralStr]: ...
+    @overload
+    def normcase(self, path: PathLike[AnyStr]) -> AnyStr: ...
+    @overload
+    def normcase(self, path: AnyOrLiteralStr) -> AnyOrLiteralStr: ...
+    def isabs(self, path: StrOrBytesPath) -> bool: ...
+
+class PurePathBase:
+    parser: ClassVar[types.ModuleType | ParserBase]
+    def __init__(self, path: StrPath, *paths: StrPath) -> None: ...
+    def with_segments(self, *pathsegments: StrPath) -> Self: ...
+    def as_posix(self) -> str: ...
+    @property
+    def drive(self) -> str: ...
+    @property
+    def root(self) -> str: ...
+    @property
+    def anchor(self) -> str: ...
+    @property
+    def name(self) -> str: ...
+    @property
+    def suffix(self) -> str: ...
+    @property
+    def suffixes(self) -> list[str]: ...
+    @property
+    def stem(self) -> str: ...
+    def with_name(self, name: str) -> Self: ...
+    def with_stem(self, stem: str) -> Self: ...
+    def with_suffix(self, suffix: str) -> Self: ...
+    def relative_to(self, other: StrPath, *, walk_up: bool = False) -> Self: ...
+    def is_relative_to(self, other: StrPath) -> bool: ...
+    @property
+    def parts(self) -> tuple[str, ...]: ...
+    def joinpath(self, *pathsegments: StrPath) -> Self: ...
+    def __truediv__(self, key: StrPath) -> Self: ...
+    def __rtruediv__(self, key: StrPath) -> Self: ...
+    @property
+    def parent(self) -> Self: ...
+    @property
+    def parents(self) -> Sequence[Self]: ...
+    def is_absolute(self) -> bool: ...
+    def match(self, path_pattern: str, *, case_sensitive: bool | None = None) -> bool: ...
+    def full_match(self, pattern: StrPath, *, case_sensitive: bool | None = None) -> bool: ...
+
+class PathBase(PurePathBase):
+    def stat(self, *, follow_symlinks: bool = True) -> stat_result: ...
+    def lstat(self) -> stat_result: ...
+    def exists(self, *, follow_symlinks: bool = True) -> bool: ...
+    def is_dir(self, *, follow_symlinks: bool = True) -> bool: ...
+    def is_file(self, *, follow_symlinks: bool = True) -> bool: ...
+    def is_mount(self) -> bool: ...
+    def is_symlink(self) -> bool: ...
+    def is_junction(self) -> bool: ...
+    def is_block_device(self) -> bool: ...
+    def is_char_device(self) -> bool: ...
+    def is_fifo(self) -> bool: ...
+    def is_socket(self) -> bool: ...
+    def samefile(self, other_path: StrPath) -> bool: ...
+    def open(
+        self,
+        mode: str = "r",
+        buffering: int = -1,
+        encoding: str | None = None,
+        errors: str | None = None,
+        newline: str | None = None,
+    ) -> IO[Any]: ...
```

Comment:
This should probably use the same overloads as `open()`/`Path.open()` does. But since this is a new class, we should leave out the `mode: str` fallback.